sway-core/src/semantic_analysis/symbol_resolve.rs

```diff
@@ -394,7 +394,7 @@ impl ResolveSymbols for TypeAliasDeclaration {
 
 impl ResolveSymbols for GenericArgument {
     fn resolve_symbols(&mut self, handler: &Handler, ctx: SymbolResolveContext) {
-        if let Some(call_path) = self.call_path_tree_mut() {
+        if let Some(call_path) = self.as_type_argument_mut().unwrap().call_path_tree.as_mut() {
```

Comment:
If we introduce a method mention above to remove repeating code, than this instance would actually be the `_mut` version of that method.